diff --git a/src/setuptools_github/tools.py b/src/setuptools_github/tools.py
index a436a8b..0db2ec7 100644
--- a/src/setuptools_github/tools.py
+++ b/src/setuptools_github/tools.py
@@ -306,7 +306,7 @@ def get_data(
         mod = loadmod(record_path)
         gdata = {
             "ref": mod.ref,
-            "sha": mod.sha,
+            "sha": mod.hash,
             "run_number": mod.build,
             "run_id": mod.runid,
         }
