diff --git a/src/setuptools_github/_build.py b/src/setuptools_github/_build.py
index f4175da..d4f35ea 100644
--- a/src/setuptools_github/_build.py
+++ b/src/setuptools_github/_build.py
@@ -1,8 +1,17 @@
 # autogenerate build file
+# gdata
 branch = 'refs/tags/release/0.3.10'
 build = '23'
 current = '0.3.10'
 hash = '2aa89736f1c56bc6a807764c2ab41386cf131bd9'
+ref = 'release/0.3.10'
+run_id = '5904419528'
+run_number = '23'
 runid = '5904419528'
+sha = '2aa8973'
 version = '0.3.10'
 workflow = 'tags'
+# data
+runid = '5904419528'
+version = '0.3.10'
+workflow = 'release/0.3.10'
diff --git a/src/setuptools_github/tools.py b/src/setuptools_github/tools.py
index 6e8ff4c..87a7f8b 100644
--- a/src/setuptools_github/tools.py
+++ b/src/setuptools_github/tools.py
@@ -405,7 +405,7 @@ def process(
                 yield (name, value)
 
     record_path = (Path(version_file).parent / record).absolute() if record else None
-    data, _ = get_data(version_file, github_dump, record_path, abort)
+    data, gdata = get_data(version_file, github_dump, record_path, abort)
     set_module_var(version_file, "__version__", data["version"])
     set_module_var(version_file, "__hash__", data["hash"])
 
@@ -420,7 +420,16 @@ def process(
         record_path.parent.mkdir(parents=True, exist_ok=True)
         with record_path.open("w") as fp:
             print("# autogenerate build file", file=fp)
-            for key, value in sorted((data or {}).items()):
+            print("# gdata", file=fp)
+            for key, value in sorted((gdata or {}).items()):
+                value = f"'{value}'" if isinstance(value, str) else value
+                print(f"{key} = {value}", file=fp)
+
+            print("# data", file=fp)
+            for key in sorted(["version", "hash", "runid", "workflow"]):
+                if key not in data:
+                    continue
+                value = data[key]
                 value = f"'{value}'" if isinstance(value, str) else value
                 print(f"{key} = {value}", file=fp)
 
